01_install_requirements.sh

```diff
@@ -40,9 +40,13 @@ elif [[ $DISTRO == "rhel8" ]]; then
     # Enable EPEL for python3-passlib and python3-bcrypt required by metal3-dev-env
     sudo dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
 
-    # The packaged 2.x ansible is too old for compatibility with metal3-dev-env
-    sudo dnf erase -y ansible
-    sudo subscription-manager repos --disable=ansible-2-for-rhel-8-x86_64-rpms
+    # RHEL 8 running on CI instances does not have ansible repo subscribed
+    # and installed. Thus, we can do ansible clean process for only local runs.
+    if [[ -z "$OPENSHIFT_CI" ]]; then
+      # The packaged 2.x ansible is too old for compatibility with metal3-dev-env
+      sudo dnf erase -y ansible
+      sudo subscription-manager repos --disable=ansible-2-for-rhel-8-x86_64-rpms
```

Comment:
Perhaps a more general solution would be to only perform these actions if we find a packaged ansible installed?

IIRC I only added these to allow folks using the previous packaged ansible to upgrade to later dev-scripts (that uses a pip installed version), so we could even consider removing this since #1301 which introduced this landed several months ago, so everyone should have migrated to the pip ansible by now?

Reply:
You are right @hardys.

More general solution looks elegant. But this check still might be useful(in case ansible is enabled by default) for the first dev-script installers.

I changed by checking the ansible repo is enabled in subscription-manager, and if yes, deletion process works.